100-level-AsyncAPI-Intro/step2.md

```diff
@@ -0,0 +1,244 @@
+
+Imagine that Chan's spaceship communicates with Eve on Capuccinova through various servers. In AsyncAPI, the `servers` section describes the servers our API will communicate with, including their hosts, protocols, and descriptions.
+
+```yml
+servers:
+  production:
+    host: space.mosquitto.org
+    protocol: mqtt
+    description: Production server for space communication
+```
+
+- **Channels (Communication channels in the API):**
+
+The spaceship's radio stations, which correspond to specific topics like communicating with Capuccinova, are simmilar to the `channels` in AsyncAPI. These channels represent paths for sending specific types of messages. Each channel uses an address, which is a template for the URI of the server, and can have variables inside curly braces.
+
+```yml
+channels:
+  roomidchat:
+    address: chat/{roomId}
+    messages:
+      chatMessage:
+        name: ChatMessage
+        payload:
+          type: object
+          properties:
+            userId:
+              type: string
+              description: The ID of the user who sent the message
+            message:
+              type: string
+              description: The chat message
+            spacename:
+              type: string
+              description: The unique spacename identifier for the message
+```
+
+- **Tags (Categorizing operations):**
+
+`Tags` in AsyncAPI are like the labels on the spaceship's control panel. They help categorize different operations, making it easier for developers to understand what each part of the API does. Each tag consists of a name and an optional description.
+
+```yml
+tags:
+  - name: chat
+    description: Operations related to chat
+```
+
+- **Components (Reusable parts of the API):**
+
+`Component` is like the toolbox in Chan's spaceship. It contains reusable parts like message definitions and schemas that can be used across the API. Each component is linked to a particular schema, which describes the structure of the message content.
+
+```yml
+components:
+  messages:
+    ChatMessage:
+      name: ChatMessage
+      title: Chat Message
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/ChatMessagePayload'
+  schemas:
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
+
+## Programming Operations, Messages, and Schemas
+
+- **Operations (Defined actions within channels):**
+
+`Operations` in AsyncAPI are like the instructions for the spaceship. They define what actions can be performed within each channel, such as sending or receiving messages. Each operation is associated with a message and a schema.
+
+```yml
+operations:
+  sendChatMessage:
+    action: 'send'
+    summary: Send a chat message to a specific room
+    channel:
+      $ref: '#/channels/roomidchat'
+```
```

Comment:
```suggestion
- **Operations**

`Operations` in AsyncAPI are like the instructions for the spaceship. They define what actions can be performed within each channel, such as sending or receiving messages. Each operation is associated with a message and a schema.

```yml
operations:
  sendChatMessage:
    action: 'send'
    summary: Send a chat message to a specific room
    channel:
      $ref: '#/channels/roomidchat'
```
```